org/mozilla/jss/util/NativeProxy.java

```diff
@@ -205,4 +205,32 @@ public synchronized static void assertRegistryEmpty() {
             logger.debug("NativeProxy registry is empty");
         }
     }
+
+    /**
+     * Unsafe: Purges all NativeProxies from memory.
+     *
+     * In the rare instances where we wish to shutdown an existing
+     * CryptoManager, all native proxies need to be cleared and freed.
+     * This will result in any lingering references to stop working, but
+     * should ensure that an application can recover from this scenario.
+     */
+    public synchronized static void unsafePurgeRegistry() throws Exception {
```

Comment:
Instead of "unsafe" should we call it forceClose() or something like that?

Reply:
Sure, works for me.

Reply:
I've changed this to `purgeAllInRegistry`.